src/contexts/DelegatesContext.tsx

```
@@ -18,35 +18,43 @@ export type Delegate = {
 }
 
 export interface IDelegatesContext {
+  isLoading: boolean
   delegates: Delegate[]
   getDelegateByAddress: (address: string) => Delegate | undefined
+  getDelegateByName: (name: string) => Delegate | undefined
 }
 
 const DelegatesContext = createContext<IDelegatesContext | undefined>(undefined)
 
 const DelegateContextProvider = ({ children }: DelegatesContextProps) => {
   const { network } = useNetwork()
   const [delegates, setDelegates] = useState<Delegate[]>([])
+  const [isLoading, setIsLoading] = useState<boolean>(true)
 
   useEffect(() => {
-    const fetchOpenPRs = async () => {
-      const response = await (
-        await fetch(
-          network === 'polkadot' || network === 'polkadot-lc'
-            ? DelegateListPolkadot
-            : DelegateListKusama,
-        )!
-      ).json()
+    if (!network) return
+    setIsLoading(true)
+    const networkToFetch =
+      network === 'polkadot' || network === 'polkadot-lc'
+        ? DelegateListPolkadot
+        : DelegateListKusama
 
-      const randomized = shuffleArray(response) as Delegate[]
-      setDelegates(randomized)
-    }
-    fetchOpenPRs()
+    fetch(networkToFetch)
+      .then(async (response) => {
+        const result = await response.json()
+        const randomized = shuffleArray(result) as Delegate[]
+        setDelegates(randomized)
+        setIsLoading(false)
+      })
+      .catch(console.error)
```

should we throw a popup error here?

---

There was no error handling before, and I doubt there'll ever be issues with a github hosted static content. Should there be, we'll be able to see in the console now at least, but I wouldn't bother much.